### PR TITLE
feat: allow only using macOS SDKs in CI

### PR DESCRIPTION
**Pull request by codebytere**
Allow only using macOS SDKs in CI, so that we don't need to download all of XCode. We can then leverage the `SDKROOT` env var to tell the XCode already in Actions runners which SDK to use.

Xcode SDKS for 14 and 13 are [in Azure](https://portal.azure.com/#view/Microsoft_Azure_Storage/ContainerMenuBlade/~/overview/storageAccountId/%2Fsubscriptions%2F3974e2bb-1561-4b5b-872c-162de614763d%2FresourceGroups%2Felectron-build-tools%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Felectronbuildtools/path/xcode/etag/%220x8DA26F3A767F5EB%22/defaultEncryptionScope/%24account-encryption-key/denyEncryptionScopeOverride~/false/defaultId//publicAccessVal/Blob).

We should eventually extend this for all users but this allows it for the GHA migration in the short term.

### CODE REVIEW

File: src/e-build.js [hunk 1]
```diff
@@ -102,7 +103,11 @@ program
       }
 
       if (process.platform === 'darwin') {
-        loadXcode({ target, quiet: true });
+        if (process.env.CI) {
```
Comment (VerteDinde) [inline src/e-build.js:106]: Do we also want to do this here?:
```suggestion
        if (process.env.CI || options.useSdk) {
```
Not sure if build-tools would try to trigger the Xcode download here if it didn't exist

Comment (codebytere) [reply src/e-build.js:106]: oops yep - thanks!

File: src/e-build.js [hunk 2]
```diff
@@ -102,7 +112,11 @@ program
       }
 
       if (process.platform === 'darwin') {
-        loadXcode({ target, quiet: true });
+        if (process.env.CI || options.useSdk) {
```
Comment (jkleinsc) [inline src/e-build.js:115]:
```suggestion
        if (options.useSdk) {
```
Same as above

File: src/e-init.js [hunk 3]
```diff
@@ -194,7 +199,11 @@ program
 
       // ensure xcode is loaded
       if (process.platform === 'darwin') {
-        loadXcode(true);
+        if (process.env.CI || options.useSdk) {
```
Comment (jkleinsc) [inline src/e-init.js:202]:
```suggestion
        if (options.useSdk) {
```
Same as above

**Review by jkleinsc — CHANGES_REQUESTED** [review]
This feature shouldn't turn on automatically for CI.

**Review by jkleinsc — APPROVED** [review]

**Review by ckerr — APPROVED** [review]

**Pull request merged**